### PR TITLE
Improve documentation

### PR DESCRIPTION
**Pull request by flavio**
This is the first step required to get our docs nicely integrated into [upstream one](https://www.terraform.io/docs/index.html).

This is a step toward fixing issue https://github.com/dmacvicar/terraform-provider-libvirt/issues/138

This is how the documentation would look into the official website.

## Main providers page

The change is not into this PR, it will be done against https://github.com/hashicorp/terraform/ later by me.

01 - providers page

## Main page

This is the page describing the provider

02 - main page

## Cloud init

The cloud init resource

03 - cloudinit

## Ignition

The ignition resource. Includes example featuring integration with the [official ignition](https://www.terraform.io/docs/providers/ignition/index.html) resource.

04 - ignition

## Domain

This page is **huge**. I tried to split it into sections and put cross links where it made sense.

05 - domain

## Network

Network resource

06 - network

## Volume

Volume resource

07 - volume

## Final considerations

The contents are basically a 1:1 of what we currently have. I just improved the style, made small improvements and added cross links.

There are resources where I think we should add more examples (the network one for example). But I plan to add more examples to this git repository under the `examples` directory.

What do you think, should we extend the online examples?

### CODE REVIEW

File: website/docs/r/domain.html.markdown [hunk 1]
```diff
@@ -177,53 +149,75 @@ resource "libvirt_domain" "my_machine" {
...
-* `addresses` - (Optional) An IP address for this domain in this network
-* `hostname` - (Optional) A hostname that will be assigned to this domain resource in this network.
-* `wait_for_lease`- (Optional) When creating the domain resource, wait until the network
-interface gets a DHCP lease from libvirt, so that the computed ip addresses will be available
-when the domain is up and the plan applied.
+* `addresses` - (Optional) An IP address for this domain in this network.
+* `hostname` - (Optional) A hostname that will be assigned to this domain
+  resource in this network.
+* `wait_for_lease`- (Optional) When creating the domain resource, wait until the
+  network interface gets a DHCP lease from libvirt, so that the computed ip
```
Comment (monstermunchkin) [inline website/docs/r/domain.html.markdown]: "IP" (upper case)

File: website/docs/r/domain.html.markdown [hunk 2]
```diff
@@ -276,18 +279,24 @@ resource "libvirt_domain" "my_machine" {
...
-* You can repeat the `console` block to create more than one console, in the same way
-that you can repeat `disk` blocks (see above)
-* The `target_type` is optional for the first console
-* All subsequent `console` blocks must specify a `target_type` of `virtio`
-* The `source_path` is optional for all consoles
+~> **Note well:**
+  <ul>
+    <li>You can repeat the `console` block to create more than one console, in the
+        same way that you can repeat `disk` blocks (see [above](#handling-disks)).</li>
+    <li>The `target_type` is optional for the first console.</li>
```
Comment (monstermunchkin) [inline website/docs/r/domain.html.markdown]: We should state the default value for the first console: "[...], and defaults to _whatever_."

File: website/docs/r/volume.html.markdown [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+as based on it. This way the image will not be modified and no extra disk space
+is going to be used for the base image.
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) A unique name for the resource, required by libvirt.
+  Changing this forces a new resource to be created.
+* `pool` - (Optional) The pool where the resource will be created.
```
Comment (monstermunchkin) [inline website/docs/r/volume.html.markdown]: We could be more specific and write "The storage pool [...]"

File: website/docs/r/coreos_ignition.html.markdown [hunk 4]
```diff
@@ -36,3 +38,35 @@ The following arguments are supported:
   * A rendered Terraform Ignition object
 
 Any change of the above fields will cause a new resource to be created.
+
+## Integration with Ignition provider
+
+The `libvirt_ignition` resource can be integrated with terraform
```
Comment (monstermunchkin) [inline website/docs/r/coreos_ignition.html.markdown]: We shouldn't mix _Terraform_ and _terraform_. I think we should stick with _Terraform_.

File: website/docs/r/domain.html.markdown [hunk 5]
```diff
@@ -24,56 +24,34 @@ resource "libvirt_domain" "default" {
...
+* `memory` - (Optional) The amount of memory in MiB. If not specified the domain
+  will be created with 512 MiB of memory be used.
+* `running` - (Optional) Use `false` to turn off the instance. If not specified,
+  true is assumed and the instance, if stopped, will be started at next apply.
+* `disk` - (Optional) An array of one or more disks to attach to the domain. The
+  `disk` object structure is documented [below](#handling-disks).
+* `network_interface` - (Optional) An array of one or more network interfaces to
+  attach to the domain. The `network_interface` object structure is documented
+  [below](#handling-network-interfaces).
 * `metadata` - (Optional) A string containing arbitrary data. This is going to be
```
Comment (flavio) [inline website/docs/r/domain.html.markdown]: Yes, that's something I wanted to talk about. We are not using it. I think I introduced that for a quick POC with salt a long time ago.

@moio: do you use domain metadata with sumaform?

File: website/docs/r/domain.html.markdown (hunk 2 above)
Comment (flavio) [inline website/docs/r/domain.html.markdown]: I took a quick look at the code and I couldn't understand the default type. Do you know it? You wrote the code 😄

File: website/docs/r/coreos_ignition.html.markdown (hunk 4 above)
Comment (flavio) [inline website/docs/r/coreos_ignition.html.markdown]: Fixed, dunno why github doesn't understand that.

File: website/docs/r/domain.html.markdown (hunk 2 above)
Comment (monstermunchkin) [inline website/docs/r/domain.html.markdown]: It apparently defaults to "serial".

File: website/docs/r/domain.html.markdown (hunk 5 above)
Comment (flavio) [inline website/docs/r/domain.html.markdown]: I checked their project, this is not being used. I'm going to drop it from the docs.

**Comment by flavio** [comment]
@monstermunchkin: I applied all the changes you proposed. If you are fine with them I'll squash the commits into a single one and merge everything into master.

**Pull request merged**